dubbo-common/src/main/java/org/apache/dubbo/common/extension/Inject.java

```diff
@@ -22,8 +22,20 @@
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+import static org.apache.dubbo.common.extension.Inject.InjectType.ByName;
+
 @Documented
 @Retention(RetentionPolicy.RUNTIME)
 @Target({ElementType.TYPE, ElementType.METHOD})
-public @interface DisableInject {
+public @interface Inject {
+    // whether enable injection or not
+    boolean enable() default true;
+
+    // inject type default by name injection
+    InjectType type() default ByName;
+
+    enum InjectType{
+        ByName,
+        ByType
+    }
```

Comment:
What is purpose of changing this annotation

Reply:
Hi AlbumenJ, DisableInject isn't generic, this Inject annotation is more generic, 
then ExtensionLoader can't support inject by type,  so refactor it, that's all!

Reply:
I am worried about this change may break some implementation work fine before commit.

Reply:
First of all all ut&it pass, then this pr is compatible with legacy code, 
I think break implementation do not exist!